src/heka/sandbox.c

```diff
@@ -1058,24 +1059,40 @@ const char* lsb_heka_get_lua_file(lsb_heka_sandbox *hsb)
 
 lsb_heka_stats lsb_heka_get_stats(lsb_heka_sandbox *hsb)
 {
-  if (!hsb) return (struct lsb_heka_stats){ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+  if (!hsb) return (struct lsb_heka_stats){ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
 
   return (struct lsb_heka_stats){
-    .mem_cur      = lsb_usage(hsb->lsb, LSB_UT_MEMORY, LSB_US_CURRENT),
-    .mem_max      = lsb_usage(hsb->lsb, LSB_UT_MEMORY, LSB_US_MAXIMUM),
-    .out_max      = lsb_usage(hsb->lsb, LSB_UT_OUTPUT, LSB_US_MAXIMUM),
-    .ins_max      = lsb_usage(hsb->lsb, LSB_UT_INSTRUCTION, LSB_US_MAXIMUM),
-    .im_cnt       = hsb->stats.im_cnt,
-    .im_bytes     = hsb->stats.im_bytes,
-    .pm_cnt       = hsb->stats.pm_cnt,
-    .pm_failures  = hsb->stats.pm_failures,
-    .pm_avg       = hsb->stats.pm.mean,
-    .pm_sd        = lsb_sd_running_stats(&hsb->stats.pm),
-    .te_avg       = hsb->stats.te.mean,
-    .te_sd        = lsb_sd_running_stats(&hsb->stats.te)
+    .mem_cur     = lsb_usage(hsb->lsb, LSB_UT_MEMORY, LSB_US_CURRENT),
+    .mem_max     = lsb_usage(hsb->lsb, LSB_UT_MEMORY, LSB_US_MAXIMUM),
+    .ext_mem_cur = lsb_usage(hsb->lsb, LSB_UT_EXTERNAL_MEMORY, LSB_US_CURRENT),
+    .ext_mem_max = lsb_usage(hsb->lsb, LSB_UT_EXTERNAL_MEMORY, LSB_US_MAXIMUM),
+    .out_max     = lsb_usage(hsb->lsb, LSB_UT_OUTPUT, LSB_US_MAXIMUM),
+    .ins_max     = lsb_usage(hsb->lsb, LSB_UT_INSTRUCTION, LSB_US_MAXIMUM),
+    .im_cnt      = hsb->stats.im_cnt,
+    .im_bytes    = hsb->stats.im_bytes,
+    .pm_cnt      = hsb->stats.pm_cnt,
+    .pm_failures = hsb->stats.pm_failures,
+    .pm_avg      = hsb->stats.pm.mean,
+    .pm_sd       = lsb_sd_running_stats(&hsb->stats.pm),
+    .te_avg      = hsb->stats.te.mean,
+    .te_sd       = lsb_sd_running_stats(&hsb->stats.te)
   };
 }
 
+size_t lsb_heka_get_ext_memory_limit(lsb_heka_sandbox *hsb)
```

Comment:
I don't believe this is needed as the adjustment should manage and report back an OOM condition.  Also they could retrieve it through the stats function so it is a bit redundant.